### PR TITLE
[Feature details] Big cleanup of feature details / bottom sheet impl

### PR DESCRIPTION
**Pull request by gino-m**
This includes several changes organized by commits, including a large rename commit, and a refactor and simplification of bottom sheet metrics. Sorry it's a big one, please feel free to review commits separately if it's useful; I've tried to make each commit clear and meaningful.

Fixes #463, #464.

### CODE REVIEW

**Comment by gino-m** [comment]
@dturner gentle ping to PTAL once you're back. Thanks in advance!

**Comment by dturner** [comment]
Thanks for the nudge will try to review this today or tomorrow.

File: gnd/src/main/java/com/google/android/gnd/ui/home/featuredetails/FeatureDetailsChromeBehavior.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+import android.content.Context;
+import android.util.AttributeSet;
+import android.view.View;
+import android.view.ViewGroup;
+import androidx.coordinatorlayout.widget.CoordinatorLayout;
+import com.google.android.gnd.R;
+import com.google.android.gnd.ui.home.BottomSheetDependentBehavior;
+import com.google.android.gnd.ui.home.BottomSheetMetrics;
+
+public class FeatureDetailsChromeBehavior extends BottomSheetDependentBehavior<ViewGroup> {
```
Comment (dturner) [inline gnd/src/main/java/com/google/android/gnd/ui/home/featuredetails/FeatureDetailsChromeBehavior.java:28]: Add high-level summary of what this class does

File: gnd/src/main/java/com/google/android/gnd/ui/home/featuredetails/FeatureDetailsChromeBehavior.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  public FeatureDetailsChromeBehavior(Context context, AttributeSet attrs) {
+    super(context, attrs);
+  }
+
+  @Override
+  protected void onBottomSheetChanged(
```
Comment (dturner) [inline gnd/src/main/java/com/google/android/gnd/ui/home/featuredetails/FeatureDetailsChromeBehavior.java:34]: Consider adding comment here to make it clear when this method will be called.

File: gnd/src/main/java/com/google/android/gnd/ui/home/featuredetails/FeatureDetailsFragment.java [hunk 3]
```diff
@@ -14,52 +14,49 @@
...
-import com.google.android.gnd.ui.home.FeatureSheetState;
+import com.google.android.gnd.ui.home.BottomSheetState;
 import com.google.android.gnd.ui.home.HomeScreenViewModel;
 import javax.inject.Inject;
 
-public class FeatureSheetFragment extends AbstractFragment {
-
-  @BindView(R.id.feature_header_icon)
-  ImageView featureHeaderIcon;
+public class FeatureDetailsFragment extends AbstractFragment {
```
Comment (dturner) [inline gnd/src/main/java/com/google/android/gnd/ui/home/featuredetails/FeatureDetailsFragment.java:35]: Add comment explaining what `FeatureDetails` or reference another class so it's easy to find out what object this `Fragment` is representing

File: gnd/src/main/java/com/google/android/gnd/ui/home/featuredetails/FeatureDetailsChromeBehavior.java (hunk 2 above)
Comment (gino-m) [reply gnd/src/main/java/com/google/android/gnd/ui/home/featuredetails/FeatureDetailsChromeBehavior.java:34]: Added more JavaDoc to the base class. Is that sufficient / better than duplicating JavaDoc in multiple places? Note that the IDE is able to help here.

File: gnd/src/main/java/com/google/android/gnd/ui/home/featuredetails/FeatureDetailsFragment.java (hunk 3 above)
Comment (gino-m) [reply gnd/src/main/java/com/google/android/gnd/ui/home/featuredetails/FeatureDetailsFragment.java:35]: Adding JavaDoc here. In a follow-up PR, I hope to either move the header into this fragment, in which case its existence would be more reasonably justified. (It currently only contains the observation list).

**Comment by gino-m** [comment]
@dturner Thanks for the review! PTAL for reapproval?

**Pull request merged**